### PR TITLE
SOAR-1394: SEP blacklist file

### PR DESCRIPTION
**Pull request by mrinehart-r7**
### Description

Describe the proposed changes:

  - Adds blacklist file action

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://komand.github.io/python/style.html)

- [ ] For dependencies, pin [OS package](https://komand.github.io/python/style.html#dockerfile) and [Python package](https://komand.github.io/python/style.html#requirements-txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://komand.github.io/python/sdk.html#sdk-versions) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://komand.github.io/python/error_handling.html#plugin-exceptions) and [ConnectionTestException](https://komand.github.io/python/error_handling.html#connection-exceptions)
- [ ] For logging, use [self.logger](https://komand.github.io/python/sdk.html#logging)
- [ ] For docs, use [changelog style](https://komand.github.io/python/style.html#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://komand.github.io/python/style.html#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR.
4. Add UI screenshot of the workflow used for testing
5. Add UI screenshot of the job output used for testing
6. Add UI screenshot of the artifact (See rules in UI Checklist) used for testing

### CODE REVIEW

File: symantec_endpoint_security/help.md [hunk 1]
```diff
@@ -37,6 +37,44 @@ Example input:
...
+|name|string|None|True|Name for the blacklist|None|Insightconnect Blacklist|
+
+Example input:
+
+```
+{
+  "description": "Hashes banned via InsightConnect",
+  "domain_id": "0AF740760A0414711FAA4F8BD5293158",
+  "hashes": ["9de5069c5afe602b2ea0a04b66beb2c0"],
+  "name": "Insightconnect Blacklist"
```
Comment (jschipp-r7) [inline symantec_endpoint_security/help.md:60]:
```suggestion
  "name": "Hash Blacklisted from InsightConnect"
```

File: symantec_endpoint_security/plugin.spec.yaml [hunk 2]
```diff
@@ -816,3 +816,38 @@ actions:
...
+        description: "Hashes (MD5 or SHA256) to add to the blacklist. Note: only one type of hash is allowed at a time"
+        type: "[]string"
+        required: true
+        example: '["9de5069c5afe602b2ea0a04b66beb2c0"]'
+      description:
+        title: "Description"
+        description: "Description for the blacklist"
+        type: string
+        required: true
+        example: "Hashes banned via InsightConnect"
```
Comment (jschipp-r7) [inline symantec_endpoint_security/plugin.spec.yaml:834]: This is what we're using in the other vendor plugins

```suggestion
        example: "Hash Blacklisted from InsightConnect"
```

File: symantec_endpoint_security/plugin.spec.yaml [hunk 3]
```diff
@@ -816,3 +816,38 @@ actions:
         description: "Agent matching the search"
         type: agent
         required: false
+  blacklist:
+    title: "Blacklist"
+    description: "Blacklist MD5 or SHA256 hashes"
+    input:
+      hashes:
+        title: "Hashes"
+        description: "Hashes (MD5 or SHA256) to add to the blacklist. Note: only one type of hash is allowed at a time"
```
Comment (jschipp-r7) [inline symantec_endpoint_security/plugin.spec.yaml:825]: Any reason we don't automatically detect the hashes and make the appropriate requests vs requiring the user to make sure they're all the same type?

File: symantec_endpoint_security/plugin.spec.yaml [hunk 4]
```diff
@@ -816,3 +816,38 @@ actions:
...
+        description: "Description for the blacklist"
+        type: string
+        required: true
+        example: "Hashes banned via InsightConnect"
+      name:
+        title: "Name"
+        description: "Name for the blacklist"
+        type: string
+        required: true
+        example: "Insightconnect Blacklist"
```
Comment (jschipp-r7) [inline symantec_endpoint_security/plugin.spec.yaml:840]:
```suggestion
        example: "InsightConnect Blacklist"
```

File: symantec_endpoint_security/help.md (hunk 1 above)
Comment (mrinehart-r7) [reply symantec_endpoint_security/help.md:60]: This doesn't fit, the action allows for blacklisting multiple hashes into a single list - this "name" property is the name of the list. Are you okay with "Hashes Blacklisted from InsightConnect"?

File: symantec_endpoint_security/plugin.spec.yaml (hunk 3 above)
Comment (mrinehart-r7) [reply symantec_endpoint_security/plugin.spec.yaml:825]: The plugin does that already. SEP is an "one or the other" type of API where you cannot mix MD5 and SHA256 hashes in the same list, hence the description

File: symantec_endpoint_security/help.md (hunk 1 above)
Comment (jschipp-r7) [reply symantec_endpoint_security/help.md:60]: Yeah, Hashes.. looks good 👍

File: symantec_endpoint_security/plugin.spec.yaml (hunk 3 above)
Comment (jschipp-r7) [reply symantec_endpoint_security/plugin.spec.yaml:825]: Yeah, I mean. Break the list into md5[] and sha256[] and send each.

Comment (jschipp-r7) [reply symantec_endpoint_security/plugin.spec.yaml:825]: We could do this in a later version if we need to. As it's not necessary for the workflows but wanted to call this out.

Comment (mrinehart-r7) [reply symantec_endpoint_security/plugin.spec.yaml:825]: Yeah we can revisit options after MVP - getting the solution right may be a bit of work around the UX for the user and the resulting UX in the Symantec console.

**Pull request merged**